### PR TITLE
Enable libsodium version 1.0.16

### PR DESCRIPTION
**Pull request by mpw-wwu**
The version functions of libsodium were removed and replaced by constants.

### CODE REVIEW

**Comment by mpw-wwu** [comment]
Instead of checking the availability of sodium for halite, I leave it up to halite to check for the correct version of sodium. I think this will cause less trouble. I haven't tested that on older php versions though. Someone might test it at least on php7.0.

Introduced the usage of HiddenStrings if they are available. The assumption is, if they are available, halite will require it's usage.

As the code is right now, we can still give an array, a String or directly a HiddenString to encrypt(). Over time we can use HiddenStrings in the rest of the code aswell.

File: application/Library/Crypto.php [hunk 1]
```diff
@@ -40,6 +40,14 @@ protected static function getKey() {
 		return self::$key;
 	}
 
+	private static function doWeNeedHiddenStrings() : bool {
```
Comment (cyriltata) [inline application/Library/Crypto.php]: Maybe follow the existing function declaration style for consistency

Comment (mpw-wwu) [reply application/Library/Crypto.php]: You mean because of the return type declaration ”: bool“?

**Comment by cyriltata** [comment]
Ok I will test with the PHP version in production. This is abit sensitive
as could break the whole email functionality so will take a while

best
Cyril

File: application/Library/Crypto.php (hunk 1 above)
Comment (cyriltata) [reply application/Library/Crypto.php]: yes.

Comment (mpw-wwu) [reply application/Library/Crypto.php]: This is new to PHP7.0. If we're ready to drop PHP5.6 support from now on, I think this improves code quality.

Comment (cyriltata) [reply application/Library/Crypto.php]: yes you are right... it improves, just that it looks weird now to be used only in one method in a the whole class.

Comment (mpw-wwu) [reply application/Library/Crypto.php]: True, but you have to start somewhere ;). If you don't like it, we can remove it. Anyway, as you wrote below, we should test this first.

Comment (mpw-wwu) [reply application/Library/Crypto.php]: I just noticed, if we drop PHP7 support, this is obligatory anyway and we don't need to check for it. If we keep PHP5.6 support, this code doesn't run. So the way I created the PR doesn't make sense in any case.

Comment (mpw-wwu) [reply application/Library/Crypto.php]: I think @rubenarslan should have the final call on dropping PHP5.6 support. I'd support this decision, as we're not aware of any formr instances running on such an outdated system at the moment and no one will start a new instance on such an outdated version.

Comment (rubenarslan) [reply application/Library/Crypto.php]: I'm pro. Let's drop it, before someone decides to run formr on 5.6 ;-)

Comment (mpw-wwu) [reply application/Library/Crypto.php]: Is has been dropped in the documentation now. So if you agree, we can start implementing new functions with predefined return types.

**Comment by mpw-wwu** [comment]
As discussed in last week's developers meeting: We wait to merge this until we find a way to migrate old keys, else we'll see database corruptions.

**Pull request merged**